src/com/luissolano/Main.java

```diff
@@ -203,6 +216,16 @@ void drain() {
                     }
                 }
             }
+
+            @Override
+            public void request(long l) {
+                s.request(l);
+            }
+
+            @Override
+            public void cancel() {
+                s.cancel();
+            }
```

Comment:
Not sure either if these `request` and `cancel` methods are correct. I guess not, since this operator may reduce the number of events in the stream.